src/components/common/FPaginate.vue

```diff
@@ -0,0 +1,166 @@
+                :disable="disable"
+                class="paginate-select"
+                dense
+                borderless
+                @input="setPerPage"
+            />
+        </div>
+        <div class="col-auto">
+            <q-pagination
+                :value="value.page"
+                :max="pageMax"
+                :disable="disable"
+                input
+                @input="setPage"
+            />
+        </div>
+    </div>
+</template>
+
+<script>
+    import _ from 'lodash';
+    import UtilMixin from 'src/mixins/utils';
+
+    export default {
+        name: 'FPaginate',
+
+        mixins: [
+            UtilMixin,
+        ],
+
+        props: {
+            value: {
+                type: Object,
+                default: () => ({})
+            },
+            noPerPage: {
+                type: Boolean,
+                default: false,
+            },
+            asNeeded: {
+                type: Boolean,
+                default: false,
+            },
+            alignEnd: {
+                type: Boolean,
+                default: false,
+            },
+            disable: {
+                type: Boolean,
+                default: false,
+            },
+            perPageText: {
+                type: String,
+                default: ''
+            },
+            totalItemsText: {
+                type: String,
+                default: ''
+            },
+            showItemsTotal: {
+                type: Boolean,
+                default: false,
+            },
+        },
+
+        data() {
+            let perPageOptions = [10, 30, 50];
+
+            return {
+                perPageOptions: this.arrayToOptions(perPageOptions),
+            };
+        },
+
+        computed: {
+            isXs() {
+                return this.$q.screen.xs;
+            },
+
+            pageMax() {
+                return Math.ceil(this.value.total / this.value.perPage) || 1;
+            },
+
+            checkPerPageMessage() {
+                return this.perPageText
+                    ? this.perPageText
+                    : 'results-per-page';
+            }
+        },
+
+        methods: {
+            setPerPage(newPerPage) {
+                const newPagination = _.cloneDeep(this.value);
+
+                newPagination.page = 1;
+                newPagination.perPage = newPerPage.value;
+                this.$emit('input', newPagination);
+                this.$emit('change', { perPage: newPerPage.value });
+            },
+
+            setPage(newPage) {
+                const newPagination = _.cloneDeep(this.value);
+
+                newPagination.page = newPage;
+                this.$emit('input', newPagination);
+                this.$emit('change', { page: newPage });
+            },
+
+            arrayToOptions(array) {
+                return _.map(array, element => {
+                    return {
+                        label: `${element}`,
```

Comment:
Don't we have util for this?

Reply:
Now using from utils, also here array contains numbers only so not handling translations.